doc/glossary/index.rst

```diff
@@ -31,3 +31,374 @@ Glossary
+        specific tokeninfo and set user-defined tokeninfos.
+
+   Token
+        All token information is stored in an SQL database, while you may choose, which database
+        you want to use. privacyIDEA uses SQLAlchemy to map the database to internal objects.
+        Thus you may choose to run privacyIDEA with SQLite, MySQL, PostgreSQL, Oracle,
+        DB2 or other database.
+
+        PrivacyIDEA supports a great variety of different token types.
+        They each have different requirements concerning configuration and how
+        the authentication works. This chapter explains the authentication modes, lists the
+        supported hardware and software tokens and explains how the token types can be used
+        with privacyIDEA.
+        Tools which facilitate and automate token enrollment are found in :ref:`enrollment_tools`.
+
+        * :ref:`authentication_modes`
+        * :ref:`supported_tokens`
+        * :ref:`tokentypes`
+
+   Audit
+        The systems provides a sophisticated audit log, that can be viewed in the WebUI.
+
+        The Audit log lists all events the server registers.
+
+        .. figure:: /audit/auditlog.png
+           :width: 500
+
+           *Audit Log*
+
+        privacyIDEA comes with a default SQL audit module (see :ref:`code_audit`).
+
+        Starting with version 3.2 privacyIDEA also provides a :ref:`logger_audit` and
+        a :ref:`container_audit` which can be used to send privacyIDEA audit log messages
+        to services like splunk or logstash.
+
+   Tokenowner
+        The owner of a token is the user for whom the token was rolled out.
+
+   FailCount
+   MaxFail
+        The FailCount count the number of failed login attempts.
+
+        If the login fail counter reaches the ``MaxFail`` the user can not login
+        with this token anymore.
+
+        The administrator or help desk user can select those tokens and
+        click the button ``reset failcounter`` to reset the fail counter to zero.
+        The tokens can be used for authentication again.
+
+   SplitAtSign
+        ``splitAtSign`` defines if the username like *user@company*
+        given during authentication should
+        be split into the loginname *user* and the realm name *company*.
+        In most cases this is the wanted behaviour so this is enabled by default.
+
+        But given your users log in with email addresses like *user@gmail.com* and
+        *otheruser@outlook.com* you probably do not want to split.
+
+        How a user is related to a realm is described here: :ref:`relate_realm`
+
+        This option also affects the login via the :ref:`rest_auth`
+
+   Rollout State
+        A token can be rolled out in several steps like the 2step HOTP/TOTP token.
+        In this case the attribute ``“rollout_state”`` of the token contains certain values
+        like ``‘clientwait’ or ‘enrolled’``. This way actions can be triggered, depending
+        on the step during an enrollment process.
```

Comment:
Rollout States are:     

'clientwait'
 # The rollout is pending in the backend, like CSRs that need to be approved
'pending'
 # This means the user needs to authenticate to verify that the token was successfully enrolled.
'verify'
'enrolled'
'failed'

Note, not all tokens have the rollout state "enrolled" set consistently. An empty rollout state means "enrolled".

Reply:
done in 88b3cb4811dc144e9bdec74d2e232fa843076b2d.

Reply:
See and fix it.

Reply:
done in 9e1730c84ea3c457b4b8ed7ec1956ef9876a77ed.